### PR TITLE
Fix: set arrival time in millis (not seconds) (#766)

### LINKED ISSUE #766: `setArrivalTime` publishes both in seconds and milliseconds
`setArrivalTime` publishes both in seconds and milliseconds

### Expected behaviour
arrivalTime should be published only as milliseconds since epoch

### Actual behavior
arrivalTime is published  as both seconds and milliseconds since epoch 

source 1: [src/main/java/com/iota/iri/network/Node.java](https://github.com/iotaledger/iri/blob/d59de9fbaa2432a7123d9a4d65a1555241fcd934/src/main/java/com/iota/iri/network/Node.java#L392)
source 2: [src/main/java/com/iota/iri/service/API.java](https://github.com/iotaledger/iri/blob/bc6b943368695ab266db31347c1752ba79dca846/src/main/java/com/iota/iri/service/API.java#L660)

### Am I planning to do it myself with a PR?
No.

### PR DESCRIPTION
**Pull request by georgmittendorfer**
# Description

In API method storeTransactionStatement arrival date was set in seconds. Changed that to milliseconds. See cliri bug fix referenced in issue.

Fixes #766 
Replaces PR #1369

* Added tests
* Updated mockito (for new tests)
* Refactored incompatible mockito usage
* Made Iota.configuration private (instance access not needed and should be injected. Was easier to use this way in tests, too)
* Added documentation that arrival time is in millis

Attention: SnapshotServiceImpl#isOrphaned NOT changed. See comment in issue.

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

# How Has This Been Tested?

Unit tests that check that set arrival time is close to System.currentTimeMillis

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/test/java/com/iota/iri/service/APITest.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import java.util.Collections;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.longThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
```
Comment (GalRogozinski) [inline src/test/java/com/iota/iri/service/APITest.java]: Hmm, now that you have upgraded Mockito this doesn't give a warning.

I agree that it is nicer than then using `@rule`, the problem is that we started using rules elsewhere in the code.
@kwek20, what do you think? You want to make the switch?

File: src/test/java/com/iota/iri/service/tipselection/impl/WalkerAlphaTest.java [hunk 2]
```diff
@@ -58,8 +58,7 @@ public static void setUp() throws Exception {
         tangle.init();
 
         TailFinder tailFinder = Mockito.mock(TailFinder.class);
-        Mockito.when(tailFinder.findTail(Mockito.any(Hash.class)))
-                .then(args -> Optional.of(args.getArgumentAt(0, Hash.class)));
+        Mockito.when(tailFinder.findTail(Mockito.any(Hash.class))).then(args -> Optional.of(args.getArgument(0)));
```
Comment (GalRogozinski) [inline src/test/java/com/iota/iri/service/tipselection/impl/WalkerAlphaTest.java]: I actually like it that the line was broken down. I found it more readable.
I need to play with the `format_settings.epf` file so it won't format the lines this way.

**Comment by georgmittendorfer** [comment]
`For next time though if you want to do refactoring that is not in the scope of the issue please put it in a different PR.` -> OK! I knew you would be saying this, as we had the same discussion last time... :sweat_smile:

Reason (assuming you mean Iota.configuration): I implemented against an old revision (there it was in scope) but someone refactored the code in the meantime. After merging these changes  were left over and I was too lazy to revert and move them into another branch.

File: src/test/java/com/iota/iri/service/APITest.java (hunk 1 above)
Comment (georgmittendorfer) [reply src/test/java/com/iota/iri/service/APITest.java]: Did this out of habit. Just tell me what you prefer an I will change it. Another option would be not to use the runner at all and call `MockitoAnnotations.initMocks(this);` in the `@Before`. That way the tests stay open for other Runners.

File: src/test/java/com/iota/iri/service/tipselection/impl/WalkerAlphaTest.java (hunk 2 above)
Comment (georgmittendorfer) [reply src/test/java/com/iota/iri/service/tipselection/impl/WalkerAlphaTest.java]: OK, will change that.

File: src/test/java/com/iota/iri/service/APITest.java (hunk 1 above)
Comment (kwek20) [reply src/test/java/com/iota/iri/service/APITest.java]: I think using the `@Rule` is fine for now. If we ever decide to make a change for Junit 5, we will need to do an overhaul anyways, and we can decide then.

If you want to change to a different runner, `@RunWith` is available.
Please change it to the following:

```
@Rule
public MockitoRule mockitoRule = MockitoJUnit.rule();
```

Comment (georgmittendorfer) [reply src/test/java/com/iota/iri/service/APITest.java]: @kwek20 OK, will do!

**Comment by georgmittendorfer** [comment]
There is something wrong with the new unit tests. Only some builds fail with:

`[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.112 s <<< FAILURE! - in com.iota.iri.network.NodeTest
[ERROR] whenProcessReceivedDataSetArrivalTimeToCurrentMillis(com.iota.iri.network.NodeTest)  Time elapsed: 0.073 s  <<< ERROR!
java.lang.NullPointerException
	at com.iota.iri.network.NodeTest.whenProcessReceivedDataSetArrivalTimeToCurrentMillis(NodeTest.java:88)`

I had this once locally and it made `continuous-integration/travis-ci/pr — The Travis CI build failed` fail, too. WIll investigate but can't reproduce. Any ideas?

**Pull request merged**